### ISSUE TITLE
[Feature request] Add a shortcut to toggle between the incremental search and the main window

### ISSUE THREAD

**Issue by Kimpa-Tammas** [notepad-plus-plus/notepad-plus-plus#13126]
Currently, a shortcut exists to open the incremental search window. But there is now way to take again the focus to the main window other than mouse or close the incremental search window by the Escape key. When this window is focused, all other shortcuts don't work. I think that a shortcut or a specific way without using mouse is necessary to take again the focus to the main window.

![Sans titre-2](https://user-images.githubusercontent.com/33954530/218808803-78b02979-e3c2-41bf-843f-df832051c40a.png)

**Comment by alankilborn** [comment]
You could tap the Windows key twice. :-)

**Comment by Kimpa-Tammas** [comment]
Thank you for the tip, but my request is about a specific and customizable shortcut.

**Comment by Kimpa-Tammas** [comment]
Thank you for this tip! I already created a script for this, but your is better.

**Comment by alankilborn** [comment]
@Kimpa-Tammas What is your suggestion for the keycombination to move back to the editor window when the _Incremental Search_ window is focused?

**Comment by Kimpa-Tammas** [comment]
> @Kimpa-Tammas What is your suggestion for the keycombination to move back to the editor window when the _Incremental Search_ window is focused?

@alankilborn: I use currently the Esc key, but this shortcut close the incremental search. I think that the best behavior should be like the regular search, for example: I select a word or nothing and I start the incremental search from the menu or from 
the shortcut. At this time, the incremental search windows remains open and the focus returns to the editor. Like F3, a shortcut will be useful to go to the next (Or previous) incremental search result without to click on the arrows.

**Comment by alankilborn** [comment]
@Kimpa-Tammas Sorry but I don't understand that answer to my question.  I was considering to fix this issue in Notepad++, and needed a direct answer to my question to proceed.

>I use currently the Esc key, but this shortcut close the incremental search

I understand this part, but since this relates to closing the Incremental Search and not merely switching back to the editor window, it isn't relevant (to my question).

>Like F3, a shortcut will be useful to go to the next (Or previous) incremental search result without to click on the arrows.

I recently fixed this to make F3 and Shift+F3 work while Incremental Search window has input focus.  But this isn't relevant to my question either, because for F3 to work for incremental search, input focus has to remain with the Incremental Search window, not shift back to the editing window.

If F3 shifted focus from Incremental Search back to the editor, the next F3 would act on the data in the regular Find window, so if you were hinting that F3 should shift input focus back to the editor, this really doesn't make sense.

**Comment by Kimpa-Tammas** [comment]
Sorry, I didn't understand your question. English is not my native language. I didn't mean to use F3 (Or Shift-F3) for incremental search. It was an example. I explained to don't close the incremental search and get the editor window focus after launched the incremental search. And then, use some shortcuts to switch between occurrences without activate again the incremental search window. It's just a suggestion, but someone will be maybe a better way.

**Comment by alankilborn** [comment]
OK, well, I wanted someone else (besides me) to say that the same key that invokes _Incremental Search_ from the editor window (default:  Ctrl+Alt+i) should also return input focus to the editor window when pressed when the _Find:_ control (the edit box) has input focus.

This would be much like how F7 moves focus from the editing window to the _Search results_ window, and then F7 in that window moves focus back to the editing window.

**Comment by alankilborn** [comment]
> I explained to don't close the incremental search and get the editor window focus after launched the incremental search. And then, use some shortcuts to switch between occurrences without activate again the incremental search

This goes beyond the original issue, but perhaps not a bad idea.

Comparing again to the _Search results_ window, this would be like what F4 and Shift+F4 do from the editing window to move from match to match.

**Comment by Kimpa-Tammas** [comment]
Indeed, a similar way to the search results should be nice. In my opinion, the focus change isn't the first priority, but above all add shortcuts for incremental search arrow buttons. I'm sure many people use incremental search and should give their opinion on the subject.

**Comment by alankilborn** [comment]
>the focus change isn't the first priority

But that's the subject of this issue.

>add shortcuts for incremental search arrow buttons

F3 and Shift+F3 are for this, but ONLY when the Find box of the Incremental Search window has focus.  I wonder if there would be acceptance of these keys to function this way from the editing window:

* if user has most-recently-used the (normal) _Find_ window for searching, F3 finds the next match for that window's _Find what_ box data (same as pressing _Find_ window's _Find Next_ button)
* if user has most-recently-used the _Incremental Search_ window for searching, F3 finds the next match for that window's _Find_ box data (same as pressing _Incremental Search_ window's _>_ button)

**Comment by alankilborn** [comment]
Maybe @vinsworldcom has opinions on any or all aspects of this issue, as a known (by me) _Incremental Search_ devotee.

**Comment by vinsworldcom** [comment]
I use `ESC` to change focus back to the main window.  And with @alankilborn  latest fix to make `F3` find next work and the match count script, I see no need to keep incremental search window open when I'm not "actively" using it.  And by "actively" I mean pressing `Enter` to go to the next match.  When I find where I want to be (presumably to edit), I press `ESC` and focus goes to editor and I start editing.

Cheers.

**Comment by alankilborn** [comment]
I was planning to work on `Problem*3` under a separate issue that I have yet to create.

**Comment by alankilborn** [comment]
@xypha In your `Behavior N*1`, step 4, I can't replicate F3/Shift+F3 behavior described.  Note that I tried replicating starting from step 1 right after I start up Notepad++.

**Comment by xypha** [comment]
@Kimpa-Tammas
Updated the code from the workaround posted [earlier](https://github.com/notepad-plus-plus/notepad-plus-plus/issues/13126#issuecomment-1854261788)

```ahk
; updated 2024.11.07 and 2025.09.19
#Requires AutoHotkey v2.0

#HotIf WinActive("ahk_class Notepad++")

*^s::           ; Ctrl + s          = save / save all
^w::            ; Ctrl + w          = close tab
*^Tab::         ; Ctrl + Tab        = switch tab forwards/backwards
^g::            ; Ctrl + G          = open Go To dialogue box
^h::            ; Ctrl + H          = open Find-Replace dialogue box
^+f::           ; Ctrl + Shift + F  = open Find dialogue box
{
CloseIncrSearch(ThisHotkey)
}

CloseIncrSearch(key) {
DetectHiddenText False  ; only check visible text
If WinActive("ahk_class Notepad++", "Find:") and ControlGetClassNN(ControlGetFocus("A")) !== "Scintilla1" {
; If incremental search is present (i.e. "Find:" text is visible) and has keyboard focus
    Send "^a"       ; CTRL + A ; select all text typed in search field     ~~ comment out if not needed
    Send "^c"       ; CTRL + C ; copy search term to clipboard             ~~ comment out if not needed
    Send "{Esc}"    ; close Incremental Search
    Sleep 100       ; wait for closure/focus main window                   ~~ comment out if not needed
    }
DetectHiddenText True                                   ; restore default
key := RegExReplace(key, "([a-zA-Z]{2,})" , "{$1}")     ; add {} around words like Tab, Up...
Send StrReplace(key, "*")                               ; Send "hotkey" after removing wildcard modifier
; Sleep 100       ; wait for dialogue to open and focus           ~~ uncomment if ^v needed
; Send "^v"       ; CTRL + V ; paste search term from clipboard   ~~ uncomment if needed

/* alternative to above
If ControlGetVisible("✕", "ahk_class Notepad++")
    ControlClick("✕", "ahk_class Notepad++")

Send RegExReplace(key, "([a-zA-Z]{2,})" , "{$1}") 
*/
}

#HotIf
```

Updated code prevents AHK from focusing and closing the incremental search if it was not already focused.
If incremental search is not focused, then the "If" section is skipped and dialogue box is directly opened (without closing incremental search).
I thought I'd share this because it worked a bit better and can be applied to several shortcuts at the same time.
